add steps for pulling staging data in windows (#502)
wagtail/wagtail.org: README.md

fab pull-production-media
```

### Pulling staging data

> If you'd like to work with staging data and have access, run the following commands if you're on Mac/Linux (POSIX):

Reviewer: What does 'have access' mean?

I believe it means you have been granted administration access to Heroku. **Suggestion:** let's make that more clear.

Author: Right! I agree with this.

> If you'd like to work with staging data and have access, run the following commands if you're on Mac/Linux (POSIX):

Reviewer 2: ```suggestion
If you'd like to work with staging data and have been granted access, run the following commands if you're on Mac/Linux (POSIX):
```

```
fab pull-staging-data
```

Access will only be given when absolutely necessary.

> If you're on Windows, we recommend you follow these [steps.](#pulling-staging-data-on-windows)

Reviewer: **Suggestion:** for accessibility, the link text should make the purpose of the link clear. Just 'steps' on its own could imply many things, I think 'specific steps for Windows' makes the purpose much more clear.

```suggestion
If you're on Windows, we recommend you follow the [specific steps for Windows](#pulling-staging-data-on-windows).
```

For more info see [Frontend general info](docs/frontend/general-info.md)

## Pulling staging data on Windows

> The `fab pull-staging-data` command does not work on Windows. Windows seems to bark at the way quotes in the commands are handled in the `fabfile.py` file. You will have to run the pertinent `fabfile.py` commands manually. These are the `dexec()` commands, and they are of three types:

Reviewer: **Suggestion:** stating that the command don't work on Windows and that the instructions that follow are workaround is enough.

```suggestion
The `fab pull-staging-data` command does not currently work on Windows due to the way quotes are handled. 

The recommended workaround is to run the necessary commands manually. 
```

1. The ones to be run on the `web` container.
2. The ones to be run on the `db` container.

> 3. The ones to be run directly on my local terminal.

Reviewer: Your _personal_ terminal? 😉 

This, and the section after, is written from your personal perspective (`my terminal`; `I used`).

Try writing with an imperative tone of voice, like a series of instructions for a recipe. 

`Mix the ingredients in a bowl` --> `run this command in your terminal`

Reviewer 2: Storm makes a good point about style. If this were a tutorial for the Wagtail blog then using a more personalized style would be okay. For documentation, we should aim to be less personal and more precise.

Author: 😅 Yes!
I'll admit I wasn't as thorough, especially because I wasn't even sure if we would use the steps directly as they were. The general idea was finding a spot in the `README.md`. 

I'll be sure to implement these new suggestions, though.

> 3. The ones to be run directly on my local terminal.

Reviewer 2: ```suggestion
3. The ones to be run directly on a local terminal.
```

2. The ones to be run on the `db` container.
3. The ones to be run directly on a local terminal.

> You will need two separate terminal instances for the `web` and local. We suggest using the VS code editor, and the terminal in it. The `db` container instructions should be done from within the Docker terminal.

Reviewer: Should we expand on this? How do you get a shell inside the `db` container?

Author: I'm open to suggestions on whether we should. In my case, I just click on the running `db` container and it opens it by default. Under the `exec` tab.

> You will need two separate terminal instances for the `web` and local. We suggest using the VS code editor, and the terminal in it. The `db` container instructions should be done from within the Docker terminal.

Reviewer 2: ```suggestion
You will need two separate terminal instances for the `web` and local. We suggest using the VS code editor, and the terminal in it. The `db` container instructions should be done from within the Docker Desktop terminal because that is one of the easiest ways to activate a shell within Docker.
```

> The ones for the web container and local machine are somewhat more straightforward to run. I used the VS code editor, and the terminal in it. More than one terminal instance is required for the `web` container and local commands, but the db container instructions should be done from within the Docker terminal.

Reviewer 2: ```suggestion
The ones for the web container and local machine are somewhat more straightforward to run. We suggest using the VS code editor, and the terminal in it. More than one terminal instance is required for the `web` container and local commands, but the db container instructions should be done from within the Docker terminal.
```

The commands, the location, and the order in which to run them are as follows:

1. `heroku pg:backups:download --output=here.dump --app wagtail-org-staging`. To avoid less typing, Substitute the `original{LOCAL_DUMP_DIR}/{datestamp}.dump}` with `here.dump`. Also substitute the `app_instance` variable with the literal `wagtail-org-staging`. Run this on your local terminal instance.
2. The next command is to be run within the `db` container, so open it and type `dropdb --if-exists --host db --username=wagtailorg wagtailorg`.
3. Afterwards, still in the `db` container, run `createdb --host db --username=wagtailorg wagtailorg`.
4. The next command involves connecting the local postgres DB using `psql`. For this, run the following in the `db` container: `psql -d wagtailorg -U wagtailorg -c "CREATE SCHEMA heroku_ext;"`
5. Still in the db container, run the following command for importing the database file to the `db` container: `pg_restore --clean --no-acl --if-exists --no-owner --host db --username=wagtailorg -d wagtailorg /app/here.dump `. This is possible if you also named the dump file `"here.dump"` as I did in step 1.

> 6. The next command to run is to be in the web container (in one of the open terminals on possibly your VS code editor, after previously running `make sh`). It is: `psql -c "UPDATE wagtailcore_site SET hostname = 'localhost', port = 8000 WHERE is_default_site IS TRUE;"`.

Reviewer 2: ```suggestion
6. The next command to run is to be in the web container (which is likely in one of the open terminals of your VS code editor, after previously running `make sh`). It is: `psql -c "UPDATE wagtailcore_site SET hostname = 'localhost', port = 8000 WHERE is_default_site IS TRUE;"`.
```

7. The last command to run is in the local terminal, and it is `rm here.dump`.

> With `npm` installed, you can run `npm install` directly, since nvm isn't supported for Windows (non-POSIX). Optionally, you can use `fnm` to first switch to the appropriate version of node.

Reviewer 2: ```suggestion
With `npm` installed, run `npm install` directly since nvm isn't supported for Windows (non-POSIX). Optionally, `fnm` can be used to first switch to the appropriate version of node.
```

Afterwards, the recommended next step would be setting up your [frontend tooling.](#frontend-tooling-docker-and-vagrant)

> ## Deployment

Reviewer: Have you considered using [`docker compose run db <command to run in container>`](https://docs.docker.com/reference/cli/docker/compose/run/)? I think this makes it much easier for users to run commands in a container. You can just list a series of commands to run to accomplish whatever `fab pull-staging-data` is doing.

Overall, my impression of this section is that it is overly 'dense' (too many words) compared to other sections of the README.

Author: I tried this, and it worked up until step 4, then things went downhill. The commands would only run when in the docker terminal. Step 5 also failed to run anywhere else but in the docker terminal.

Reviewer: Alright, no worries
